ratis-client/src/main/java/org/apache/ratis/client/impl/DataStreamClientImpl.java

```diff
@@ -69,11 +70,6 @@
     private final RaftClientRequest header;
     private final CompletableFuture<DataStreamReply> headerFuture;
     private final CompletableFuture<RaftClientReply> raftClientReplyFuture = new CompletableFuture<>();
-    private final MemoizedSupplier<CompletableFuture<DataStreamReply>> closeSupplier = JavaUtils.memoize(() -> {
```

Comment:
We need some way to make the close idempotent.  May be a AtomicBoolean.

Reply:
@szetszwo Hi, I have updated the patch, but confused why we need Atomic, I think will not happen multi-thread close the same stream.